### DISCUSSION TITLE
programming problem on gps time server

### DISCUSSION THREAD

**Discussion by alalal80** [Montecri/GNSSTimeServer#14] [General]
Good evening,
 When programming I got this error:

Compilation error: index_bytearray.h: No such file or directory
Could you give me the right bookstores?

**Comment by Montecri** [comment]
Hi.

[index_bytearray.h](https://github.com/Montecri/GNSSTimeServer/blob/main/src/index_bytearray.h) is in the "src" folder, as well as other files necessary (ota.h and main.cpp).

Please note that now the project offers pre-compiled firmware for ESP8266 and ESP32 on [Releases](https://github.com/Montecri/GNSSTimeServer/releases) section.
Regards,

- Cristiano

**Comment by alalal80** [comment]
Good evening,
 thank you, indeed it works better, but I have the following error:

Compilation error: 'class RtcDateTime' has no member named 'Unix32Time'; did you mean 'Epoch32Time'?


----- Mail original -----
De: "Cristiano Monteiro" ***@***.***>
À: "Montecri/GNSSTimeServer" ***@***.***>
Cc: "alalal80" ***@***.***>, "Author" ***@***.***>
Envoyé: Mercredi 20 Mars 2024 23:26:01
Objet: Re: [Montecri/GNSSTimeServer] programming problem on gps time server (Discussion #14)





Hi. 

index_bytearray.h is in the "src" folder, as well as other files necessary (ota.h and main.cpp). 

Regards, 

    * Cristiano

**Reply by Montecri** [reply]
Hi!

Can you, please, give me the full console output?

Thanks!

- Cristiano

**Comment by alalal80** [comment]
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino: In function 'void PrintRTCstatus()':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino:900:18: error: 'class RtcDateTime' has no member named 'Unix32Time'; did you mean 'Epoch32Time'?
   time_t t = Now.Unix32Time();
                  ^~~~~~~~~~
                  Epoch32Time
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino: In function 'void SetRTC(time_t)':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino:929:13: error: 'class RtcDateTime' has no member named 'InitWithUnix32Time'; did you mean 'InitWithEpoch32Time'?
   timeToSet.InitWithUnix32Time(t);
             ^~~~~~~~~~~~~~~~~~
             InitWithEpoch32Time
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino: In function 'void SyncWithRTC()':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino:1193:21: error: 'class RtcDateTime' has no member named 'Unix32Time'; did you mean 'Epoch32Time'?
   long int a = time.Unix32Time();
                     ^~~~~~~~~~
                     Epoch32Time
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp: In function 'void PrintRTCstatus()':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp:900:18: error: 'class RtcDateTime' has no member named 'Unix32Time'; did you mean 'Epoch32Time'?
   time_t t = Now.Unix32Time();
                  ^~~~~~~~~~
                  Epoch32Time
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp: In function 'void SetRTC(time_t)':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp:929:13: error: 'class RtcDateTime' has no member named 'InitWithUnix32Time'; did you mean 'InitWithEpoch32Time'?
   timeToSet.InitWithUnix32Time(t);
             ^~~~~~~~~~~~~~~~~~
             InitWithEpoch32Time
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp: In function 'void SyncWithRTC()':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp:1193:21: error: 'class RtcDateTime' has no member named 'Unix32Time'; did you mean 'Epoch32Time'?
   long int a = time.Unix32Time();
                     ^~~~~~~~~~
                     Epoch32Time

exit status 1

Compilation error: 'class RtcDateTime' has no member named 'Unix32Time'; did you mean 'Epoch32Time'?

----- Mail original -----
De: "Cristiano Monteiro" ***@***.***>
À: "Montecri/GNSSTimeServer" ***@***.***>
Cc: "alalal80" ***@***.***>, "Author" ***@***.***>
Envoyé: Jeudi 21 Mars 2024 22:18:12
Objet: Re: [Montecri/GNSSTimeServer] programming problem on gps time server (Discussion #14)





Hi! 

Can you, please, give me the full console output? 

Thanks! 

    * Cristiano

**Comment by Montecri** [comment]
Hi!

Can you, please, tell me the following:

- Which environment are you using, Arduino IDE or PlatformIO?
- Do you have the following libraries installed at the same version or, if the version is different, which it is?:
        paulstoffregen/Time@^1.6
	makuna/RTC@^2.3.5
	olikraus/U8g2@^2.28.8
	arcao/Syslog@^2.0.0
	mikalhart/TinyGPSPlus@^1.0.3

- For which MCU are you trying to build, ESP8266 or ESP32?
    - Which board is selected on Arduino IDE/PlatformIO?
  
Thanks.

- Cristiano

**Comment by alalal80** [comment]
hi;


Which environment are you using, Arduino IDE or PlatformIO?
 Arduino IDE
    * 

Do you have the following libraries installed at the same version or, if the version is different, which it is?: paulstoffregen/Time@^1.6 makuna/RTC@^2.3.5 olikraus/U8g2@^2.28.8 arcao/Syslog@^2.0.0 mikalhart/TinyGPSPlus@^1.0.3 
I watch
    * 

For which MCU are you trying to build, ESP8266 or ESP32?  ESP32
        * Which board is selected on Arduino IDE/PlatformIO?  
Arduino IDE esp32 dev kit 1

Do you have a new version planned soon?


Thanks. 


----- Mail original -----
De: "Cristiano Monteiro" ***@***.***>
À: "Montecri/GNSSTimeServer" ***@***.***>
Cc: "alalal80" ***@***.***>, "Author" ***@***.***>
Envoyé: Vendredi 22 Mars 2024 12:38:38
Objet: Re: [Montecri/GNSSTimeServer] programming problem on gps time server (Discussion #14)





Hi! 

Can you, please, tell me the following: 

    * 

Which environment are you using, Arduino IDE or PlatformIO? 
    * 

Do you have the following libraries installed at the same version or, if the version is different, which it is?: paulstoffregen/Time@^1.6 makuna/RTC@^2.3.5 olikraus/U8g2@^2.28.8 arcao/Syslog@^2.0.0 mikalhart/TinyGPSPlus@^1.0.3 
    * 

For which MCU are you trying to build, ESP8266 or ESP32? 
        * Which board is selected on Arduino IDE/PlatformIO? 


Thanks. 

    * Cristiano

**Reply by Montecri** [reply]
Hi!

Not sure if I understand which libraries/versions you have installed. Can you, please, list to me, from the libraries I mentioned, which version you have installed for each?

About a new version: The current version is really fresh! I've added a lot of new functionalities in the past 6 months, like OTA upgrade, Boot diagnostics, Ethernet support, several bug fixes, etc. I'm studying new features and will announce when they are mature enough in my head to get my hands dirty.

Regards,

- Cristiano

**Comment by alalal80** [comment]
hi;


Which environment are you using, Arduino IDE or PlatformIO? Arduino IDE
    * 

Do you have the following libraries installed at the same version or, if the version is different, which it is?: paulstoffregen/Time@^1.6 makuna/RTC@^2.3.5 olikraus/U8g2@^2.28.8 arcao/Syslog@^2.0.0 mikalhart/TinyGPSPlus@^1.0.3 
    * 

For which MCU are you trying to build, ESP8266 or ESP32?  ESP32
        * Which board is selected on Arduino IDE/PlatformIO?  Arduino IDE esp32 dev kit 1


Thanks. 

----- Mail original -----
De: "Cristiano Monteiro" ***@***.***>
À: "Montecri/GNSSTimeServer" ***@***.***>
Cc: "alalal80" ***@***.***>, "Author" ***@***.***>
Envoyé: Vendredi 22 Mars 2024 12:38:38
Objet: Re: [Montecri/GNSSTimeServer] programming problem on gps time server (Discussion #14)





Hi! 

Can you, please, tell me the following: 

    * 

Which environment are you using, Arduino IDE or PlatformIO? 
    * 

Do you have the following libraries installed at the same version or, if the version is different, which it is?: paulstoffregen/Time@^1.6 makuna/RTC@^2.3.5 olikraus/U8g2@^2.28.8 arcao/Syslog@^2.0.0 mikalhart/TinyGPSPlus@^1.0.3 
    * 

For which MCU are you trying to build, ESP8266 or ESP32? 
        * Which board is selected on Arduino IDE/PlatformIO? 


Thanks. 

    * Cristiano

**Reply by Montecri** [reply]
You may have a library conflict on Arduino IDE.

Can you please compile it on PlatformIO?

PlatformIO is the environment I used to code, and it should be more resilient to library dependencies. The full **_platformio.ini_** file is available in the **src** directory.

Thanks & Regards,

- Cristiano

**Comment by alalal80** [comment]
Good evening,
Thank you for your information, I will try.
Sincerely.

----- Mail original -----
De: "Cristiano Monteiro" ***@***.***>
À: "Montecri/GNSSTimeServer" ***@***.***>
Cc: "alalal80" ***@***.***>, "Author" ***@***.***>
Envoyé: Vendredi 22 Mars 2024 20:46:01
Objet: Re: [Montecri/GNSSTimeServer] programming problem on gps time server (Discussion #14)





You may have a library conflict on Arduino IDE. 

Can you please compile it on PlatformIO? 

PlatformIO is the environment I used to code, and it should be more resilient to library dependencies. The full platformio.ini file is available in the src directory. 

Thanks & Regards, 

    * Cristiano

**Comment by alalal80** [comment]
Good evening,
could you send me all the libraries of the NTP server project.
Sincerely.

----- Mail original -----
De: ***@***.***
À: "Montecri/GNSSTimeServer" ***@***.***>
Envoyé: Jeudi 21 Mars 2024 22:27:26
Objet: Re: [Montecri/GNSSTimeServer] programming problem on gps time server (Discussion #14)

D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino: In function 'void PrintRTCstatus()':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino:900:18: error: 'class RtcDateTime' has no member named 'Unix32Time'; did you mean 'Epoch32Time'?
   time_t t = Now.Unix32Time();
                  ^~~~~~~~~~
                  Epoch32Time
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino: In function 'void SetRTC(time_t)':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino:929:13: error: 'class RtcDateTime' has no member named 'InitWithUnix32Time'; did you mean 'InitWithEpoch32Time'?
   timeToSet.InitWithUnix32Time(t);
             ^~~~~~~~~~~~~~~~~~
             InitWithEpoch32Time
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino: In function 'void SyncWithRTC()':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.ino:1193:21: error: 'class RtcDateTime' has no member named 'Unix32Time'; did you mean 'Epoch32Time'?
   long int a = time.Unix32Time();
                     ^~~~~~~~~~
                     Epoch32Time
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp: In function 'void PrintRTCstatus()':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp:900:18: error: 'class RtcDateTime' has no member named 'Unix32Time'; did you mean 'Epoch32Time'?
   time_t t = Now.Unix32Time();
                  ^~~~~~~~~~
                  Epoch32Time
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp: In function 'void SetRTC(time_t)':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp:929:13: error: 'class RtcDateTime' has no member named 'InitWithUnix32Time'; did you mean 'InitWithEpoch32Time'?
   timeToSet.InitWithUnix32Time(t);
             ^~~~~~~~~~~~~~~~~~
             InitWithEpoch32Time
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp: In function 'void SyncWithRTC()':
D:\Montage NTP\GNSS Time Server\GNSSTimeServer-main\GNSSTimeServer-main\src\main\main.cpp:1193:21: error: 'class RtcDateTime' has no member named 'Unix32Time'; did you mean 'Epoch32Time'?
   long int a = time.Unix32Time();
                     ^~~~~~~~~~
                     Epoch32Time

exit status 1

Compilation error: 'class RtcDateTime' has no member named 'Unix32Time'; did you mean 'Epoch32Time'?

----- Mail original -----
De: "Cristiano Monteiro" ***@***.***>
À: "Montecri/GNSSTimeServer" ***@***.***>
Cc: "alalal80" ***@***.***>, "Author" ***@***.***>
Envoyé: Jeudi 21 Mars 2024 22:18:12
Objet: Re: [Montecri/GNSSTimeServer] programming problem on gps time server (Discussion #14)





Hi! 

Can you, please, give me the full console output? 

Thanks! 

    * Cristiano

**Reply by Montecri** [reply]
Hi,

The libraries can be downloaded here:

https://github.com/paulstoffregen/Time
https://github.com/makuna/RTC
https://github.com/mikalhart/TinyGPSPlus
https://github.com/olikraus/U8g2
https://github.com/arcao/Syslog

However, I recommend you let your IDE take care of this, and if the IDE is PlatformIO, even better. Just drop the platformio.ini file supplied here in your project root folder and let PlatformIO sort it out.

Regards,

- Cristiano

**Comment by alalal80** [comment]
Thank you, I will try it.
Do you have other developments planned?
Sincerely.

----- Mail original -----
De: "Cristiano Monteiro" ***@***.***>
À: "Montecri/GNSSTimeServer" ***@***.***>
Cc: "alalal80" ***@***.***>, "Author" ***@***.***>
Envoyé: Jeudi 20 Juin 2024 23:41:31
Objet: Re: [Montecri/GNSSTimeServer] programming problem on gps time server (Discussion #14)





Hi, 

The libraries can be downloaded here: 

https://github.com/paulstoffregen/Time 
https://github.com/makuna/RTC 
https://github.com/mikalhart/TinyGPSPlus 
https://github.com/olikraus/U8g2 
https://github.com/arcao/Syslog 

However, I recommend you let your IDE take care of this, and if the IDE is PlatformIO, even better. Just drop the platformio.ini file supplied here in your project root folder and let PlatformIO sort it out. 

Regards, 

    * Cristiano

**Reply by Montecri** [reply]
Yes, I do!

Investigating viability first, before building a prototype and publishing.

**Comment by alalal80** [comment]
Good evening,
do you have the corresponding pins for an esp32?
do you have a new version of the assembly?
Sincerely.

**Comment by alalal80** [comment]
Good evening,
do you have the corresponding pins for an esp32?
do you have a new version of the assembly?
Sincerely.

**Reply by Montecri** [reply]
Hi! Good evening,

Yes. The only difference from ESP8266 is that ESP32 uses Hardware Serial 2 and PPS LED is pin 27.

Hardware Serial 2 on ESP32 means that IO16 is RX and IO17 is TX.

For the latest binaries for each build (ESP8266, ESP32 and ESP32+W5500) please check below:

https://github.com/Montecri/GNSSTimeServer/releases

Regards,

- Cristiano

**Comment by alalal80** [comment]
Good evening,
thank you for your answers,
I can now display the number of satellites,
I had to change line 1684 to 9600.
but I don't have the time setting on the display.
there are other lines of code to modify.
what is the pin of the esp32 for the wifi led?
what is the pin of the esp32 to connect the pss output of the gps?
Best regards.

**Reply by Montecri** [reply]
Changing 1684 to 9600 is ok, will not have any negative effect on operation.
PPS pin from GPS should connect to pin 12 of ESP32
Led to indicate the WiFi status should be connected to pin 14 of ESP32

Welcome!
